packages/shared/src/components/layout/MainLayoutHeader.tsx

```diff
@@ -41,6 +42,24 @@ const SearchPanel = dynamic(
     ),
 );
 
+interface StreakButtonProps {
+  isEnabled: boolean;
+  isLoading: boolean;
+  streak: UserStreak;
+}
+
+const StreakButton = ({ streak, isEnabled, isLoading }: StreakButtonProps) => {
+  if (isLoading) {
+    return <div className="h-10 w-20 rounded-12 bg-surface-float" />;
+  }
+
+  if (!isEnabled || !streak) {
+    return null;
+  }
```

Comment:
Just wondering if this should be the first one? So could technically see a loader but have no streak?

Reply:
Yeah, thought of that too, but then it will never be `isLoading` since in the hook itself, we check whether the query should be enabled or not. Hence, will not get the loading if it is disabled.